src/world/sprite.c

```diff
@@ -39,7 +52,45 @@ void create_balloon(int x, int y, int z, int colour)
  */
 void create_duck(int targetX, int targetY)
 {
-	RCT2_CALLPROC_X(0x0067440F, targetX, 0, targetY, 0, 0, 0, 0);
+	rct_sprite* sprite = create_sprite(2);
+	if (sprite != NULL)
+	{
+		sprite->duck.sprite_identifier = 2;
+		sprite->duck.var_01 = 8;
+		sprite->duck.var_14 = 9;
+		sprite->duck.var_09 = 0xC;
+		sprite->duck.var_15 = 9;
+		int ebx = scenario_rand();
+		int ebp = ebx;
+		int edi = ebx;
+		ebp >>= 8;
+		edi >>= 16;
+		edi &= 0x3F;
+		ebp &= 0x1E;
+		targetX += ebp;
+		targetY += ebp;
```

Comment:
You can refactor line 64-71 to the following. Scenario rand is cheap but CS always likes to use up all of the bits before calling it again:

```
targetX += scenario_rand() & 0x3F;
targetY += scenario_rand() & 0x1E;
```